docs/2.9.0/docs/query/pqs-user-guide.rst

```diff
@@ -505,988 +1780,2081 @@ Use a ``--config`` file to define multiple options or reflect an infrastructure-
-readers in order. The offset-based model makes the database's isolation
-level irrelevant, so the loosest model (``read uncommitted``) is not
-harmful.
-
-When querying the datastore, first consider the type of
-read consistency required. If there is no need for consistency (for example,
-reading a historical contract regardless of lifetime), you can query payload
-tables directly without any consideration of offset.
-Another example is a liveness metric query that calculates the
-number of transactions in the datastore over the past minute.
-This could be valid without considering the
-parallel-writing method.
-
-When consistency is required, the reader must be aware of the offset
-for reading. This ensures the reader doesn't also read
-further offsets that are present when their precedent events are not yet
-stored in the database.
-
-To achieve the level of consistency that you require, including
-read-consistency with other ledger data or commands you have executed,
-consider providing a function that returns the latest
-checkpoint offset:
-
-.. code-block:: none
+   .. code:: json
 
-   -- utility functions
-   create or replace function latest_checkpoint()
-   returns table ("offset" _transactions."offset"%type, ix _transactions.ix%type) as $$
-     select max(groups."offset") as "offset", max(groups."ix") as ix
-     from (SELECT ix - ROW_NUMBER() OVER (ORDER BY ix) as delta, * FROM _transactions) groups
-     group by groups.delta
-     order by groups.delta
-     limit 1;
+      {"timestamp":"2024-01-17T00:16:31","level":"INFO","location":"com.digitalasset.scribe.pipeline.pipeline.Impl:34","description":"Starting pipeline on behalf of 'Alice_1::1220ee13431ac437d454ea59d622cfc76599e0846a3caf166b4306d47b1bf83944a6'","scribe":{"application":"scribe"}}
+      {"timestamp":"2024-01-17T00:16:33","level":"INFO","location":"com.digitalasset.scribe.pipeline.pipeline.Impl:61","description":"Last checkpoint is absent. Seeding from ACS before processing transactions with starting offset '00000000000000000b'","scribe":{"application":"scribe"}}
+      {"timestamp":"2024-01-17T00:16:34","level":"INFO","location":"com.digitalasset.zio.daml.ledgerapi.package:201","description":"Contract filter inclusive of 1 templates and 0 interfaces","scribe":{"application":"scribe"}}
+      {"timestamp":"2024-01-17T00:16:35","level":"INFO","location":"com.digitalasset.scribe.pipeline.pipeline.Impl:85","description":"Continuing from offset '00000000000000000b' and index '0' until offset '00000000000000000b'","scribe":{"application":"scribe"}}
 
-   $$ language sql;
-   create or replace function first_checkpoint()
-   returns table ("offset" _transactions."offset"%type, ix _transactions.ix%type) as $$
-     select t."offset" as "offset", t."ix" as ix from _transactions t order by ix limit 1;
+   Notice you need to use ``%label{your_label}{format}`` to describe
+   Json attribute-value pair.
 
+Database Schema Evolution
```

Comment:
Shouldn't this be called something like "upgrading PQS"?

Reply:
Schema Evolution is a very specific term that refers to how you manage a schema throughout it's lifecycle - and evolve it, it's not really about upgrading though that's when it happens (?)